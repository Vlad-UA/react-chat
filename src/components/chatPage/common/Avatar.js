import React from 'react';
import MUIAvatar from '@material-ui/core/Avatar';
import PropTypes from 'prop-types';
import titleInitials from '../../../utils/title-initials';
import getColorFrom from '../../../utils/getColorFrom';

const Avatar = ({ title, lettersQuantity, colorFrom }) => (
  <MUIAvatar style={{ backgroundColor: getColorFrom(colorFrom) }}>
    {titleInitials({ title, lettersQuantity })}
  </MUIAvatar>
);

Avatar.propTypes = {
  title: PropTypes.string.isRequired,
  lettersQuantity: PropTypes.number.isRequired,
  colorFrom: PropTypes.string.isRequired,
};

export default Avatar;
