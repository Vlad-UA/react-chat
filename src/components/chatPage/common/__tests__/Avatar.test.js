/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Avatar from '../Avatar';

const mockProps = {
  title: 'title name',
  lettersQuantity: 2,
  colorFrom: 'colorFrom',
};

describe('<Avatar />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Avatar {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot renders correctly', () => {
    const tree = renderer.create(<Avatar {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
