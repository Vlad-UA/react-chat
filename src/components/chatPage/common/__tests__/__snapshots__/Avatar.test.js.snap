// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> snapshot renders correctly 1`] = `
<div
  className="MuiAvatar-root-4 MuiAvatar-colorDefault-5"
  style={
    Object {
      "backgroundColor": "#673ab7",
    }
  }
>
  TN
</div>
`;
